src/aap_eda/services/validation.py

```diff
@@ -0,0 +1,114 @@
+#  Copyright 2023 Red Hat, Inc.
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+import logging
+
+from django.core.exceptions import ValidationError
+from django.db import transaction
+from django.utils.translation import gettext_lazy as _
+
+from aap_eda.core import models
+from aap_eda.core.enums import ActivationStatus
+
+logger = logging.getLogger(__name__)
+
+
+def validate_activation(activation_id: int) -> bool:
```

Comment:
@hsong-rh Does this have to be a service, its not talking to any external server. Can't it just be a mixin that can be called at creation time and post creation.

Reply:
I agree that conceptually speaking is not a service. But why a mixin? It may make harder the re usability if we are going to use the same logic in other places, like at restarting and activation.